www/js/diary/LabelDetailsScreen.tsx

```diff
@@ -3,128 +3,113 @@
   Navigated to from the main LabelListScreen by clicking a trip card. */
 
 import React, { useContext } from "react";
-import { View, ScrollView, StyleSheet, useWindowDimensions } from "react-native";
-import { Appbar, Divider, IconButton, Surface, Text, useTheme } from "react-native-paper";
+import { View, Modal, ScrollView, useWindowDimensions } from "react-native";
+import { Appbar, Divider, Surface, Text, useTheme } from "react-native-paper";
 import { LabelTabContext } from "./LabelTab";
 import { cardStyles } from "./cards/DiaryCard";
 import LeafletView from "../components/LeafletView";
 import { useTranslation } from "react-i18next";
 import MultilabelButtonGroup from "../survey/multilabel/MultiLabelButtonGroup";
 import UserInputButton from "../survey/enketo/UserInputButton";
-import { getAngularService } from "../angular-react-helper";
-import { useImperialConfig } from "../config/useImperialConfig";
 import { useAddressNames } from "./addressNamesHelper";
+import { Icon } from "../components/Icon";
+import { SafeAreaView } from "react-native-safe-area-context";
+import useDerivedProperties from "./useDerivedProperties";
+import StartEndLocations from "./StartEndLocations";
 
 const LabelScreenDetails = ({ route, navigation }) => {
 
   const { surveyOpt, timelineMap } = useContext(LabelTabContext);
-  const { getFormattedDistance, distanceSuffix } = useImperialConfig();
   const { t } = useTranslation();
   const { height: windowHeight } = useWindowDimensions();
   const { colors } = useTheme();
-  const { tripId } = route.params;
-  const trip = timelineMap.get(tripId);
+  const trip = timelineMap.get(route.params.tripId);
+  const { displayDate, displayStartTime, displayEndTime,
+          displayTime, formattedDistance, formattedSectionProperties,
+          distanceSuffix, percentages } = useDerivedProperties(trip);
```

Comment:
These are variables, right? Or are they functions? If they are functions, I would suggest renaming to reflect that (e.g. by prefixing with `get`) in the next release.

Reply:
> For my own edification, this is because we shouldn't "fill in" the trips after they are read per the "react way of doing things", right?

Yep!

> These are variables, right? Or are they functions?

They're variables, destructured from the object returned by `useDerivedProperties(t)`